services/madoc-ts/src/frontend/site/pages/search.tsx

```diff
@@ -1,127 +1,138 @@
-    const [facetOptions, setFacetOptions] = useState<SearchFacet[]>([]);
-
-    const manageFacet = (facetType: string, facetValue: string) => {
-      const facet = {
-        type: 'metadata',
-        subtype: facetType,
-        value: facetValue,
-      };
-
-      if (!appliedFacets.find((fac: any) => fac.value === facet.value)) {
-        const newappliedFacets = [...appliedFacets];
-        newappliedFacets.push(facet);
-        setAppliedFacets(newappliedFacets);
-      } else if (appliedFacets.find((fac: any) => fac.value === facet.value)) {
-        const newappliedFacets = appliedFacets.filter(
-          (fac: any) => fac.value === facet.value && fac.facetType === facet.subtype
-        );
-        setAppliedFacets(newappliedFacets);
-      }
-    };
-
-    const mapFacets = () => {
+    const facetOptions = useMemo(() => {
       const options = [];
-      if (data && data.facets && data.facets.metadata) {
-        for (const [key, value] of Object.entries(data.facets.metadata)) {
+      if (searchResponse && searchResponse.facets && searchResponse.facets.metadata) {
+        for (const [key, value] of Object.entries(searchResponse.facets.metadata)) {
           const subtype = key;
-          for (const [k] of Object.entries(value)) {
+          for (const [k] of Object.entries(value as any)) {
             options.push({
               type: 'metadata',
               subtype: subtype,
               value: k,
-              applied: appliedFacets.find((option: any) => option.subtype === subtype && option.value === k)
-                ? true
-                : false,
+              applied: facets && facets[subtype] && facets[subtype].indexOf(k) !== -1,
             });
           }
         }
       }
-      setFacetOptions(options);
-    };
 
-    useEffect(() => {
-      mapFacets();
+      return options;
+    }, [facets, searchResponse]);
+
+    const clearFacets = () => {
+      history.push(
+        `${pathname}?${stringify({
+          fulltext,
+          page: 1,
+          madoc_id,
+        })}`
+      );
+    };
 
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [data, appliedFacets]);
+    const applyFacet = (type: string, value: string) => {
+      if (data) {
+        const newFacets = { ...(facets || {}) };
 
-    useEffect(() => {
-      const jsonFacets = JSON.stringify(appliedFacets);
-      if (appliedFacets && appliedFacets.length >= 1) {
-        history.push(`${pathname}?${stringify({ fulltext, page, madoc_id })}&facets=${jsonFacets}`);
-      } else {
-        history.push(`${pathname}?${stringify({ fulltext, page, madoc_id })}`);
+        if (!newFacets[type]) {
+          newFacets[type] = [];
+        }
+        newFacets[type].push(value);
+
+        history.push(
+          `${pathname}?${stringify({
+            fulltext,
+            page: 1,
+            madoc_id,
+            facets: JSON.stringify(newFacets),
+          })}`
+        );
```

Comment:
@ionab I think the change I made will apply the facets as soon as you select the item, instead of storing in state first. Sorry! You could hold the state inside of the `SearchFacets` component and then pass down _all_ of the facets when you call `applyFacets`

Reply:
Ok think I've got it working again now, storing the state locally.